File: packages/webview_flutter/webview_flutter_android/test/test_binary_messenger.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'dart:ui';

import 'package:flutter/services.dart';

class TestBinaryMessenger implements BinaryMessenger {
  @override
  Future<void> handlePlatformMessage(
    String channel,
    ByteData? data,
    PlatformMessageResponseCallback? callback,
  ) async {
    // Do nothing.
  }

  @override
  Future<ByteData?>? send(String channel, ByteData? message) async {
    return StandardMessageCodec().encodeMessage(<Object?, Object?>{});
  }

  @override
  void setMessageHandler(String channel, MessageHandler? handler) {
    // Do nothing.
  }
}

File: packages/webview_flutter/webview_flutter_android/android/src/test/java/io/flutter/plugins/webviewflutter/TestBinaryMessenger.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.webviewflutter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.flutter.plugin.common.BinaryMessenger;
import java.nio.ByteBuffer;

public class TestBinaryMessenger implements BinaryMessenger {
  @Override
  public void send(@NonNull String s, @Nullable ByteBuffer byteBuffer) {}

  @Override
  public void send(
      @NonNull String s, @Nullable ByteBuffer byteBuffer, @Nullable BinaryReply binaryReply) {}

  @Override
  public void setMessageHandler(
      @NonNull String s, @Nullable BinaryMessageHandler binaryMessageHandler) {}
}
